registration-processor/core-processor/registration-processor-demo-dedupe-stage/src/test/java/io/mosip/registrationprocessor/stages/demodedupe/DemodedupeProcessorTest.java

```diff
@@ -1,851 +1,989 @@
+
+		byte[] b = "sds".getBytes();
+		ReflectionTestUtils.setField(demodedupeProcessor, "demodedupeMatchDecesion", "MarkAsDemodedupeSuccess");
+		PowerMockito.mockStatic(JsonUtil.class);
+		PowerMockito.mockStatic(IOUtils.class);
+		PowerMockito.when(JsonUtil.class, "inputStreamtoJavaObject", inputStream, PacketMetaInfo.class)
+				.thenReturn(packetMetaInfo);
+		PowerMockito.when(IOUtils.class, "toByteArray", inputStream).thenReturn(b);
+		Mockito.when(registrationStatusService.getRegistrationStatus(any())).thenReturn(registrationStatusDto);
+		Mockito.when(abisHandlerUtil.getPacketStatus(any())).thenReturn(AbisConstant.PRE_ABIS_IDENTIFICATION);
+		Mockito.when(demoDedupe.performDedupe(anyString())).thenReturn(duplicateDtos);
+		Mockito.when(packetManagerService.getBiometrics(any(), any(), any(), any(), any())).thenReturn(null);
+		BiometricRecordValidationException e = new BiometricRecordValidationException("Biometirc validation failed");
+		Mockito.doThrow(e).when(abisHandlerUtil).validateBiometricRecord(any(), anyList());
+		List<String> uniqueRIDs = new ArrayList<>();
+		uniqueRIDs.add("2018701130000410092018110731");
+		uniqueRIDs.add("2018701130000410092018110732");
+		Mockito.when(abisHandlerUtil.getUniqueProcessedRecords(any(), any(), any(), anyList())).thenReturn(uniqueRIDs);
+		MessageDTO messageDto = demodedupeProcessor.process(dto, stageName);
+		assertTrue(messageDto.getIsValid());
+
+	}
+
+	@Test
+	public void testDemoDedupeNewPacketSuccessWithDuplicatesWithNoBiometricsWithDemodedupeRejected() throws Exception {
+
+		byte[] b = "sds".getBytes();
+		ReflectionTestUtils.setField(demodedupeProcessor, "demodedupeMatchDecesion", "MarkAsDemodedupeRejected");
+		PowerMockito.mockStatic(JsonUtil.class);
+		PowerMockito.mockStatic(IOUtils.class);
+		PowerMockito.when(JsonUtil.class, "inputStreamtoJavaObject", inputStream, PacketMetaInfo.class)
+				.thenReturn(packetMetaInfo);
+		PowerMockito.when(IOUtils.class, "toByteArray", inputStream).thenReturn(b);
+		Mockito.when(registrationStatusService.getRegistrationStatus(any())).thenReturn(registrationStatusDto);
+		Mockito.when(abisHandlerUtil.getPacketStatus(any())).thenReturn(AbisConstant.PRE_ABIS_IDENTIFICATION);
+		Mockito.when(demoDedupe.performDedupe(anyString())).thenReturn(duplicateDtos);
+		Mockito.when(packetManagerService.getBiometrics(any(), any(), any(), any(), any())).thenReturn(null);
+		BiometricRecordValidationException e = new BiometricRecordValidationException("Biometirc validation failed");
+		Mockito.doThrow(e).when(abisHandlerUtil).validateBiometricRecord(any(), anyList());
+		List<String> uniqueRIDs = new ArrayList<>();
+		uniqueRIDs.add("2018701130000410092018110731");
+		uniqueRIDs.add("2018701130000410092018110732");
+		Mockito.when(abisHandlerUtil.getUniqueProcessedRecords(any(), any(), any(), anyList())).thenReturn(uniqueRIDs);
+		MessageDTO messageDto = demodedupeProcessor.process(dto, stageName);

```

Comment:
possible to do a argument capture and assert on the latest transaction status code and status code ? If possible we can do the same in all the test cases.

Reply:
fixed